runtime/vm/JVMImage.cpp

```diff
@@ -74,30 +125,69 @@ JVMImage::freeSubAllocatedMemory(void* address)
 }
 
 void
-JVMImage::reallocateImageMemory(UDATA size)
+JVMImage::readImageFromFile()
 {
-	return;
+    OMRPORT_ACCESS_FROM_OMRPORT(getPortLibrary());
+
+    char imageBuffer[_imageSize];
+    memset(imageBuffer, 0, sizeof(imageBuffer));
+
+    intptr_t fileDescriptor = omrfile_open(_dumpFileName, EsOpenRead, 0666);
```

Comment:
whats the 0666? Or rather where are the modes defined in `file_open(const char *path, int32_t flags, int32_t mode)` @tajila

Reply:
Modes correspond to the chmod modes, https://en.wikipedia.org/wiki/Chmod